mobile-browser-based-version/src/helpers/training/training_manager.js

```diff
@@ -48,39 +49,44 @@ export class TrainingManager {
    * Train the task's model either alone or in a distributed fashion depending on the user's choice.
    * @param {Object} dataset the dataset to train on
    * @param {Boolean} distributedTraining train in a distributed fashion
+   * @param {personalizationType} personalizationTypeChosen the type of personalization chosen for the model.
    */
-  async trainModel(dataset, distributedTraining) {
+  async trainModel(
+    dataset,
+    distributedTraining,
+    personalizationTypeChosen = modelDefinition.personalizationType.NONE
+  ) {
     this.data = dataset.Xtrain;
     this.labels = dataset.ytrain;
 
-    /**
-     * If IndexedDB is turned on and the working model exists, then load the
-     * existing model from IndexedDB. Otherwise, create a fresh new one.
-     */
-    const modelParams = [
-      this.task.taskID,
-      this.task.trainingInformation.modelID,
-    ];
-    if (this.useIndexedDB && (await getWorkingModelMetadata(...modelParams))) {
-      this.model = await getWorkingModel(...modelParams);
-    } else {
-      this.model = await this.task.createModel();
+    switch (personalizationTypeChosen) {
+      case modelDefinition.personalizationType.NONE:
+        this.model = new modelDefinition.Model(this.task, this.useIndexedDB);
+        break;
+
+      case modelDefinition.personalizationType.INTEROPERABILITY:
+        this.model = new modelDefinition.InteroperabilityModel(
+          this.task,
+          this.useIndexedDB
+        );
+        break;
     }
+    await this.model.init();
 
     // Continue local training from previous epoch checkpoint
-    if (this.model.getUserDefinedMetadata() === undefined) {
-      this.model.setUserDefinedMetadata({ epoch: 0 });
+    if (this.model.getModel().getUserDefinedMetadata().epoch === undefined) {
+      this.model.getModel().getUserDefinedMetadata().epoch = 0;
```

Comment:
Is it possible to set this.model.getModel().getUserDefinedMetadata().epoch = 0 earlier, such that on creation it is guaranteed to be defined (then we can remove these 3 lines).

Reply:
Yup though about this as well, gonna add this in model creation.